shared-bindings/countio/Counter.c

```diff
@@ -53,7 +53,7 @@ STATIC mp_obj_t countio_counter_make_new(const mp_obj_type_t *type, size_t n_arg
     const countio_edge_t edge = validate_edge(args[ARG_edge].u_obj, MP_QSTR_edge);
     const digitalio_pull_t pull = validate_pull(args[ARG_pull].u_obj, MP_QSTR_pull);
 
-    countio_counter_obj_t *self = m_new_obj(countio_counter_obj_t);
+    countio_counter_obj_t *self = m_new_ll_obj(countio_counter_obj_t);
```

Comment:
```suggestion
    // Make long-lived because some implementations use a pointer to the object as interrupt-handler data.
    countio_counter_obj_t *self = m_new_ll_obj(countio_counter_obj_t);
```

Reply:
Made the suggested change.